### PR TITLE
[build] refine CMake files

### PR DESCRIPTION
**Pull request by wgtdkp**
This PR refines CMake files

- avoid setting `CMAKE_C_FLAGS` and `CMAKE_CXX_FLAGS` in project root directory.
- add mbedtls and libevent wrappers to inherit include directories.
- specify target sources separately and include header files as well.
- follow openthread/openthread CMake style.

### CODE REVIEW

File: src/library/CMakeLists.txt [hunk 1]
```diff
@@ -26,89 +26,89 @@
...
+    PRIVATE
+        openthread/bloom_filter.cpp
+        openthread/bloom_filter.hpp
+        openthread/crc16.cpp
+        openthread/crc16.hpp
+        openthread/pbkdf2_cmac.cpp
+        openthread/pbkdf2_cmac.hpp
+        openthread/random.cpp
+        openthread/random.hpp
+        openthread/README.md
```
Comment (gocarlos) [inline src/library/CMakeLists.txt:41]: ?

File: src/library/CMakeLists.txt [hunk 2]
```diff
@@ -26,89 +26,88 @@
...
-    mbedtls
-    mbedx509
-    mbedcrypto
-    fmt::fmt
-    event_core
-    event_pthreads
-    commissioner-common)
+target_sources(commissioner
+    PRIVATE
+        openthread/bloom_filter.cpp
```
Comment (bukepo) [inline src/library/CMakeLists.txt:33]: sort all sources?

File: third_party/mbedtls/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+## Mbedtls builds static libraries by default.
+if (BUILD_SHARED_LIBS)
+    set(USE_STATIC_MBEDTLS_LIBRARY OFF CACHE BOOL "disable static mbedtls library")
+    set(USE_SHARED_MBEDTLS_LIBRARY ON  CACHE BOOL "enable shared mbedtls libary")
+endif()
+
```
Comment (gjc13) [inline third_party/mbedtls/CMakeLists.txt:34]: Suggest to add `set(ENABLE_TESTING OFF CACHE BOOL "Disable mbedtls test")` and `set(ENABLE_PROGRAMS OFF CACHE BOOL "Disable mbetls program")`

**Comment by wgtdkp** [comment]
@gocarlos more comments?

File: third_party/mbedtls/CMakeLists.txt (hunk 3 above)
Comment (wgtdkp) [reply third_party/mbedtls/CMakeLists.txt:34]: sure.

Comment (wgtdkp) [reply third_party/mbedtls/CMakeLists.txt:34]: But notice that the whole third party directory is added with `EXCLUDE_FROM_ALL`, which makes only targets explicitly referenced are built.

File: third_party/mbedtls/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+## Mbedtls builds static libraries by default.
+if (BUILD_SHARED_LIBS)
+    set(USE_STATIC_MBEDTLS_LIBRARY OFF CACHE BOOL "disable static mbedtls library")
+    set(USE_SHARED_MBEDTLS_LIBRARY ON  CACHE BOOL "enable shared mbedtls libary")
```
Comment (gocarlos) [inline third_party/mbedtls/CMakeLists.txt:32]: libary -> library

File: src/app/CMakeLists.txt [hunk 5]
```diff
@@ -26,28 +26,42 @@
...
-    ${PROJECT_SOURCE_DIR}/third_party/json/repo/single_include)
+target_include_directories(commissioner-app
+    PUBLIC
+        ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
+target_include_directories(commissioner-app
+    PRIVATE
+        ${PROJECT_SOURCE_DIR}/third_party/json/repo/single_include
+)
```
Comment (gocarlos) [inline src/app/CMakeLists.txt:53]: I'd think that if the target is added, e.g. `nlohman_json::nlohman_json`, then no need for doing this manually

File: src/library/CMakeLists.txt [hunk 6]
```diff
@@ -26,89 +26,87 @@
...
+        mbedcrypto
+        fmt::fmt
+        event_core
+        event_pthreads
+        commissioner-common
+)
+
+target_include_directories(commissioner
+    PUBLIC ${PROJECT_SOURCE_DIR}/include
+)
```
Comment (gocarlos) [inline src/library/CMakeLists.txt:96]:
```cmake
target_include_directories(commissioner PUBLIC
  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
  $<INSTALL_INTERFACE:include>
)
```

Comment (wgtdkp) [reply src/library/CMakeLists.txt:96]: And I should not install headers manually, right?

Comment (gocarlos) [reply src/library/CMakeLists.txt:96]: yes, this would be the best approach.

**Comment by gocarlos** [comment]
LGTM 👍

**Comment by wgtdkp** [comment]
@gocarlos maybe you want to approve this change?

You can click "Files changed", then click "review changes" at right hand side.

**Pull request merged**